File: commons/javafx/src/test/java/com/cedarsoft/commons/javafx/OverrideBindingTest.java
package com.cedarsoft.commons.javafx;

import static org.assertj.core.api.Assertions.*;

import org.junit.jupiter.api.*;

import javafx.beans.property.SimpleBooleanProperty;

class OverrideBindingTest {

  @Test
  void testOverrideBinding() {
    SimpleBooleanProperty myProperty = new SimpleBooleanProperty();
    SimpleBooleanProperty aProperty = new SimpleBooleanProperty();
    SimpleBooleanProperty bProperty = new SimpleBooleanProperty();

    assertThat(myProperty.get()).isFalse();
    assertThat(aProperty.get()).isFalse();
    assertThat(bProperty.get()).isFalse();

    myProperty.bind(aProperty);
    assertThat(myProperty.get()).isFalse();

    myProperty.bind(bProperty);
    assertThat(myProperty.get()).isFalse();

    aProperty.set(true);
    assertThat(myProperty.get()).isFalse();

    bProperty.set(true);
    assertThat(myProperty.get()).isTrue();
  }
}

File: commons/javafx/src/test/kotlin/com/cedarsoft/commons/javafx/OverrideBindingTest.kt
package com.cedarsoft.commons.javafx

import javafx.beans.property.SimpleBooleanProperty
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class OverrideBindingTest {
  @Test
  fun testOverrideBinding() {
    val myProperty = SimpleBooleanProperty()
    val aProperty = SimpleBooleanProperty()
    val bProperty = SimpleBooleanProperty()
    Assertions.assertThat(myProperty.get()).isFalse
    Assertions.assertThat(aProperty.get()).isFalse
    Assertions.assertThat(bProperty.get()).isFalse
    myProperty.bind(aProperty)
    Assertions.assertThat(myProperty.get()).isFalse
    myProperty.bind(bProperty)
    Assertions.assertThat(myProperty.get()).isFalse
    aProperty.set(true)
    Assertions.assertThat(myProperty.get()).isFalse
    bProperty.set(true)
    Assertions.assertThat(myProperty.get()).isTrue
  }
}
